### PR TITLE
Task[hmi-server]: reimplement ssemq on hmi server

### LINKED ISSUE #2004: [TASK]: Reimplement SSE/mq on hmi-server
**Describe the task**
Since moving to the spring backed hmi-server we've had to temporarily remove sse and our message queue. Investigate how to work with this in our new spring server to reenable simulation service

### PR DESCRIPTION
**Pull request by dvince2**
# Description

* Leveraging the mq/sse implementation from the pantera platform.
* New messaging queue for ciemms and sciml simulations.
* Simulation messages are sent to all users to update shared project results as they come in.  Non subscribers will have messages ignored.

Resolves #2004

### CODE REVIEW

File: packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-node-julia.vue [hunk 1]
```diff
@@ -235,31 +240,36 @@ const runCalibrate = async () => {
 	}
 };
 
-const handleIntermediateResult = (message: string) => {
+/* const handleIntermediateResult = (message: string) => {
 	const parsedMessage = JSON.parse(message);
 	console.log(parsedMessage);
-};
+}; */
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-node-julia.vue:246]: you could remove it.

File: packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-node-julia.vue [hunk 2]
```diff
@@ -126,11 +132,11 @@ import InputNumber from 'primevue/inputnumber';
 import InputText from 'primevue/inputtext';
 import Dropdown from 'primevue/dropdown';
 import { Poller, PollerState } from '@/api/api';
-import { EventSourceManager } from '@/api/event-source-manager';
 import TeraSimulateChart from '@/workflow/tera-simulate-chart.vue';
 import TeraProgressBar from '@/workflow/tera-progress-bar.vue';
 import { getTimespan } from '@/workflow/util';
 import { logger } from '@/utils/logger';
+import { subscribe, unsubscribe } from '@/services/ClientEventService';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-node-julia.vue:139]:
```suggestion
import * as ClientEventService from '@/services/ClientEventService';
```

Better to then call `ClientEventService.subscribe()`

**Review by mwdchang — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/configuration/Config.java [hunk 3]
```diff
@@ -42,6 +42,11 @@ public class Config {
 	 */
 	ClientConfig clientConfig;
 
+	/**
+	 * If queues should be declared durable.  IF running Rabbit inside docker, this should be false
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/configuration/Config.java:46]:
```suggestion
	 * If queues should be declared durable. 
	 * If running Rabbit inside docker, this should be false.
```

File: packages/server/src/main/resources/application.properties [hunk 4]
```diff
@@ -83,3 +84,9 @@ tgpt.ws.url=wss://app.staging.terarium.ai/beaker_ws/
 ########################################################################################################################
 spicedb.launchmode=TLS
 spicedb.target=spicedb.staging.terarium.ai:443
+
+# RabbitMQ exchange
```
Comment (YohannParis) [inline packages/server/src/main/resources/application.properties:88]:
```suggestion
########################################################################################################################
# RabbitMQ exchange
########################################################################################################################
```

File: packages/client/hmi-client/src/services/ClientEventService.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+			// Parse the data as a ClientEvent and pass it on to the subscribers
+			const data = JSON.parse(message.data) as ClientEvent<any>;
+			if (data.type === ClientEventType.Heartbeat) {
+				lastHeartbeat = new Date().valueOf();
+				return;
+			}
+			const handlers = subscribers.get(data.type);
+			if (handlers) {
+				handlers.forEach((handler) => handler(data));
+			}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/ClientEventService.ts:51]:
```suggestion
			if (handlers) {
				handlers.forEach((handler) => handler(data));
			} else {
			      logger.info(`Message ${data.type} has no handler available`);
			}
```

**Review by YohannParis — CHANGES_REQUESTED** [review]

**Pull request merged**